src/store/modules/facility/actions.ts

```diff
@@ -509,8 +523,42 @@ const actions: ActionTree<FacilityState, RootState> = {
     } catch(error) {
       logger.error(error)
     }
-    commit(types.FACILITY_VIRTUAL_FACILITY_LIST_UPDATED , { facilities: stateFacilities.concat(facilities), total });
-  }
+    commit(types.FACILITY_VIRTUAL_FACILITY_LIST_UPDATED, { facilities: stateFacilities.concat(facilities), total });
+  },
+
+  updateVirtualFacilities({ commit }, facilities) {
+    commit(types.FACILITY_VIRTUAL_FACILITY_LIST_UPDATED, { facilities, total: facilities.length })
+  },
+
+  async fetchArchivedFacilities({ commit }) {
+    let facilities = []
+    try {
+      const resp = await FacilityService.fetchArchivedFacilities({
+        inputFields: {
```

Comment:
You can move this fetching logic to the service itself as you created a separate service for it and nothing is dynamic in condition.

Reply:
Updated sir